packages/suite/src/hooks/wallet/coinmarket/form/useCoinmarketExchangeForm.ts

```diff
@@ -1,266 +1,273 @@
-        composeRequest,
-        ...methods,
-    });
+    const filteredCexQuotes = useMemo(
+        () => getCexQuotesByRateType(rateType, innerQuotes, exchangeInfo),
+        [rateType, innerQuotes, exchangeInfo],
+    );
+    const dexQuotes = useMemo(() => innerQuotes?.filter(q => q.isDex), [innerQuotes]);
+
+    const { isComposing, composedLevels, feeInfo, changeFeeLevel, composeRequest } =
+        useCoinmarketComposeTransaction<CoinmarketExchangeFormProps>({
+            account,
+            network,
+            values: values as CoinmarketExchangeFormProps,
+            methods,
+        });
 
-    const helpers = useCoinmarketExchangeFormHelpers({
+    const helpers = useCoinmarketFormActions({
         account,
-        network,
         methods,
         setAmountLimits,
         changeFeeLevel,
         composeRequest,
+        setAccountOnChange: newAccount => {
+            dispatch(setCoinmarketExchangeAccount(newAccount));
+            setAccount(newAccount);
+        },
+    });
+
+    const { toggleAmountInCrypto } = useCoinmarketCurrencySwitcher({
+        account,
+        methods,
+        network,
+        quoteCryptoAmount: innerQuotes?.[0]?.sendStringAmount,
+        quoteFiatAmount: fiatOfBestScoredQuote ?? '',
+        inputNames: {
+            cryptoInput: FORM_OUTPUT_AMOUNT,
+            fiatInput: FORM_OUTPUT_FIAT,
+        },
     });
 
     const getQuotesRequest = useCallback(
-        async (request: ExchangeTradeQuoteRequest) => {
+        async (request: ExchangeTradeQuoteRequest, offLoading?: boolean) => {
+            setIsSubmittingHelper(!offLoading);
+
             if (abortControllerRef.current) {
                 abortControllerRef.current.abort();
             }
 
-            if (!request.send || !request.receive || !request.sendStringAmount) {
+            if (
+                !request.send ||
+                !request.receive ||
+                !request.sendStringAmount ||
+                request.sendStringAmount === '' ||
+                Number(request.sendStringAmount) === 0
+            ) {
                 timer.stop();
+                setIsSubmittingHelper(false);
 
                 return;
             }
 
             abortControllerRef.current = new AbortController();
             invityAPI.createInvityAPIKey(account.descriptor);
 
-            try {
-                const allQuotes = await invityAPI.getExchangeQuotes(
-                    request,
-                    abortControllerRef.current.signal,
-                );
+            const allQuotes = await invityAPI.getExchangeQuotes(
+                request,
+                abortControllerRef.current.signal,
+            );
 
-                return allQuotes;
-            } catch (error) {
-                console.log('Abort', error);
-            }
+            return allQuotes;
         },
         [account.descriptor, timer],
     );
 
     const getQuoteRequestData = useCallback((): ExchangeTradeQuoteRequest | null => {
```

Comment:
Consider to union check on `receiveCryptoSelect`, `sendCryptoSelect` and `sendStringAmount`. In general the validation check.